### PR TITLE
Canvas

### PR DESCRIPTION
**Pull request by iemeka**

### CODE REVIEW

File: canvas/index.html [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>HTML5 Canvas</title>
+</head>
+<body>
+<canvas id="draw" width="800" height="800"></canvas>
+<script>
+  
+  const canvas = (function(){
```
Comment (dokasto) [inline canvas/index.html]: This is weird. This is how to define a function experession
````
const canvas = () => {
   // stuff
    return stuff;
};
```

Comment (dokasto) [reply canvas/index.html]: Also, this is super confusing. You've got a function called `canvas` and another variable inside called `canvas`

File: canvas/index.html [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    const canvas = document.querySelector('#draw');
+    canvas.width = window.innerWidth;
+    canvas.height = window.innerHeight;
+    return canvas;
+  }());
+
+  const contex = (function context(){
```
Comment (dokasto) [inline canvas/index.html]: This is weird as well. Define the function as mentioned above

File: canvas/index.html [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    contex.stroke(); // draw line
+    [ctrVal.lastX, ctrVal.lastY] = [e.offsetX, e.offsetY]; // continuously updating start point after each stroke
+    ctrVal.hue++; // on each strock increase the hue to gradually fade into other colors - this show's how fast this function runs
+    ctrVal.hue = ctrVal.hue >= 360 ? 0 : ctrVal.hue; // without this it still goes round picking from start again.
+
+    flipDirection();
+    contextLineWidth();
+  }
+  
+  (function addListeners(){
```
Comment (dokasto) [inline canvas/index.html]: This function doesn't really need a name.

File: canvas/index.html (hunk 1 above)
Comment (iemeka) [reply canvas/index.html]: That's a function expression. Immediately invoked. 
changed the name of the other canvas variable.

File: canvas/index.html (hunk 2 above)
Comment (iemeka) [reply canvas/index.html]: same IIFE but doesn't need a name here

**Comment by dokasto** [comment]
@iemeka I think this is an opportunity to use classes here

**Review by dokasto — APPROVED** [review]

**Pull request merged**